File: 342-PowerOfFour/342-PowerOfFour.js
// Last updated: 10/03/2026, 10:03:20
/**
 * @param {number} n
 * @return {boolean}
 */
var isPowerOfFour = function(n) {
    if(n<=0){
        return false;
    }

    while(n%4==0){
        n=n/4;
    }
    return n==1;
};

File: 342-PowerOfFour/342-PowerOfFour.cpp
// Last updated: 13/05/2026, 13:10:32
class Solution {
public:
    bool isPowerOfFour(int n) {
          if (n <= 0)
            return false;

        while (n % 4 ==  0) {
            n = n / 4;
        }

        return n ==  1;
    }
};